src/hotspot/share/runtime/os.cpp

```diff
@@ -882,6 +882,53 @@ void os::abort(bool dump_core) {
 //---------------------------------------------------------------------------
 // Helper functions for fatal error handler
 
+// Given an address, attempt to locate both the symbol and the library it
+// resides in. If at least one of these steps was successful, prints information
+// and returns true.
+// On success prints either one of:
+// "<function name>+<offset> in <library>"
+// "<function name>+<offset>"
+// "<address> in <library>+<offset>"
+bool os::print_function_and_library_name(outputStream* st,
+                                         address addr,
+                                         char* buf, int buflen,
+                                         bool shorten_paths,
+                                         bool demangle) {
+  // If no scratch buffer given, allocate one here on stack.
+  // (used during error handling; its a coin toss, really, if on-stack allocation
+  //  is worse than (raw) C-heap allocation in that case).
+  char* p = buf;
+  if (p == NULL) {
+    p = (char*)::alloca(O_BUFLEN);
+    buflen = O_BUFLEN;
+  }
+  int offset = 0;
+  const bool have_function_name = dll_address_to_function_name(addr, p, buflen,
```

Comment:
So this function includes the arguments in the string that it returns?  The arguments make the output long and don't add anything helpful in my opinion, but it's not really worth having the code to strip them out.

Reply:
Hi Coleen, thanks for the review!

You are right. I modified the patch and removed the arguments, which did not make the code much more complex but reads better. Now it looks like this:

```
657 Signal Handlers:
658    SIGSEGV: crash_handler+0 in libjvm.so, sa_mask[0]=11100100010111111101111111111110, sa_flags=SA_RESTART|SA_SIGINFO
659     SIGBUS: crash_handler+0 in libjvm.so, sa_mask[0]=11100100010111111101111111111110, sa_flags=SA_RESTART|SA_SIGINFO
660     SIGFPE: crash_handler+0 in libjvm.so, sa_mask[0]=11100100010111111101111111111110, sa_flags=SA_RESTART|SA_SIGINFO
661    SIGPIPE: javaSignalHandler+0 in libjvm.so, sa_mask[0]=11100100010111111101111111111110, sa_flags=SA_RESTART|SA_SIGINFO
662    SIGXFSZ: javaSignalHandler+0 in libjvm.so, sa_mask[0]=11100100010111111101111111111110, sa_flags=SA_RESTART|SA_SIGINFO
663     SIGILL: crash_handler+0 in libjvm.so, sa_mask[0]=11100100010111111101111111111110, sa_flags=SA_RESTART|SA_SIGINFO
664    SIGUSR2: SR_handler+0 in libjvm.so, sa_mask[0]=00000000000000000000000000000000, sa_flags=SA_RESTART|SA_SIGINFO
665     SIGHUP: UserHandler+0 in libjvm.so, sa_mask[0]=11100100010111111101111111111110, sa_flags=SA_RESTART|SA_SIGINFO
666     SIGINT: UserHandler+0 in libjvm.so, sa_mask[0]=11100100010111111101111111111110, sa_flags=SA_RESTART|SA_SIGINFO
667    SIGTERM: UserHandler+0 in libjvm.so, sa_mask[0]=11100100010111111101111111111110, sa_flags=SA_RESTART|SA_SIGINFO
668    SIGQUIT: UserHandler+0 in libjvm.so, sa_mask[0]=11100100010111111101111111111110, sa_flags=SA_RESTART|SA_SIGINFO
669    SIGTRAP: crash_handler+0 in libjvm.so, sa_mask[0]=11100100010111111101111111111110, sa_flags=SA_RESTART|SA_SIGINFO
```